src/h3lib/lib/h3Index.c

```diff
@@ -1050,3 +1050,162 @@ H3Error H3_EXPORT(getPentagons)(int res, H3Index *out) {
  *         a Class II grid.
  */
 int isResolutionClassIII(int res) { return res % 2; }
+
+/**
+ * Validate a child position in the context of a given parent, returning
+ * an error if validation fails.
+ */
+static H3Error validateChildPos(int64_t childPos, H3Index parent,
+                                int childRes) {
+    int64_t maxChildCount;
+    H3Error sizeError =
+        H3_EXPORT(cellToChildrenSize)(parent, childRes, &maxChildCount);
+    if (NEVER(sizeError)) {
```

Comment:
In the future, we should require fuzzers be added too to try to exercise these. I am fine with merging this without the fuzzer and then adding it in a follow up PR.